org.envirocar.app/src/org/envirocar/app/views/trackdetails/TrackMapFactory.kt

```diff
@@ -0,0 +1,83 @@
+package org.envirocar.app.views.trackdetails
+
+import android.animation.ArgbEvaluator
+import org.envirocar.app.R
+import org.envirocar.core.entity.Measurement
+import org.envirocar.core.entity.Track
+import org.envirocar.map.camera.CameraUpdateFactory
+import org.envirocar.map.model.Marker
+import org.envirocar.map.model.Point
+import org.envirocar.map.model.Polyline
+import kotlin.math.max
+
+class TrackMapFactory(track: Track) {
```

Comment:
I intend to replace existing [`MapLayer`](https://github.com/enviroCar/enviroCar-app/blob/gsoc2024-map-libre-migration/org.envirocar.app/src/org/envirocar/app/views/trackdetails/MapLayer.java) & [`TrackMapLayer`](https://github.com/enviroCar/enviroCar-app/blob/gsoc2024-map-libre-migration/org.envirocar.app/src/org/envirocar/app/views/trackdetails/TrackMapLayer.java) classes with `TrackMapFactory`. Majority of logic has been kept inline with the referenced classes.

The primary aim of this class is to provide all objects required to display a track visually on a map e.g.
- Camera update.
- Start marker.
- Stop marker.
- Polyline (solid).
- Polyline (gradient).

It has been briefly used [here](https://github.com/enviroCar/enviroCar-app/blob/10a61497a8bf4e819cc4ee9fd2a3ff2a5cdd4aeb/org.envirocar.app/src/org/envirocar/app/views/tracklist/AbstractTrackListCardAdapter.kt#L187-L192) in `AbstractTrackListCardAdapter`.